client/src/app/pages/archetypes/components/archetype-form/archetype-form.tsx

```diff
@@ -0,0 +1,352 @@
+import React, { useMemo } from "react";
+import { useTranslation } from "react-i18next";
+import { AxiosError } from "axios";
+import { Control, useForm, Path } from "react-hook-form";
+import { yupResolver } from "@hookform/resolvers/yup";
+import * as yup from "yup";
+
+import {
+  ActionGroup,
+  Button,
+  ButtonVariant,
+  Form,
+} from "@patternfly/react-core";
+
+import type { Archetype, Tag } from "@app/api/models";
+import {
+  HookFormPFGroupController,
+  HookFormPFTextArea,
+  HookFormPFTextInput,
+} from "@app/components/HookFormPFFields";
+import { Autocomplete } from "@app/components/Autocomplete";
+import { NotificationsContext } from "@app/components/NotificationsContext";
+import {
+  useFetchArchetypes,
+  useFetchArchetypeById,
+  useCreateArchetypeMutation,
+  useUpdateArchetypeMutation,
+} from "@app/queries/archetypes";
+import { duplicateNameCheck, getAxiosErrorMessage } from "@app/utils/utils";
+import { useFetchTagCategories } from "@app/queries/tags";
+
+interface ArchetypeFormValues {
+  name: string;
+  description?: string;
+  comments?: string;
+  criteriaTags: string[]; // TODO: string[] only works if tags are uniquely named globally
+  tags: string[]; // TODO: string[] only works if tags are uniquely named globally
+  stakeholders?: object[];
```

Comment:
We have types for these guys already in our models file.

Reply:
I saw them.  Those should be used in the model itself, but here in the form field type, I'm planning on switching it to `string[]` just like for the tags.

Reply:
Oh right! Carry on.